gap/io.gi

```diff
@@ -761,17 +762,28 @@ function(func, s)
   od;
 
   if legacy then  # source and range are reversed
-    return func(rec(DigraphNrVertices := n,
-                    DigraphSource     := range,
-                    DigraphRange      := source));
-  fi;
-  return func(rec(DigraphNrVertices := n,
-                  DigraphRange      := range,
-                  DigraphSource     := source));
+    return DigraphNC(IsMutableDigraph,
```

Comment:
I don't have any idea why this code was originally written like this, but you should check with at least @james-d-mitchell that we're losing anything important by removing the ability of `DIGRAPHS_FromDigraph6String` to take an arbitrary "constructor", and forcing `DigraphNC`.

Reply:
Yeah, we're definitely ok to do this - getting rid of `DIGRAPHS_FromDigraph6String` was actually James's suggestion.

Reply:
I concur, it was my idea, I think this is the best solution.